ruth/simulator/segment.py

```diff
@@ -15,6 +15,9 @@ class SegmentPosition:
     # Position within the segment (in meters)
     position: LengthMeters
 
+    def __eq__(self, other):
```

Comment:
This is implemented by default for dataclasses, you don't need to implement it manually.